driver/controller.go

```diff
@@ -81,6 +88,21 @@ func (s controllerService) CreateVolume(ctx context.Context, req *csi.CreateVolu
 		return nil, status.Error(codes.InvalidArgument, err.Error())
 	}
 
+	// check if the create volume request has a data source
+	if source != nil {
+		// get the parent volumeID/snapshotID if exists
+		parentVol, parentID, err = s.validateContentSource(ctx, req)
```

Comment:
It would be better to replace parent with source in the variable names. There is currently no child/parent relationship.